packages/@react-spectrum/autocomplete/src/MobileSearchAutocomplete.tsx

```diff
@@ -227,7 +227,7 @@ const SearchAutocompleteButton = React.forwardRef(function SearchAutocompleteBut
 
   return (
     <div
-      {...mergeProps(hoverProps, focusProps, buttonProps)}
+      {...mergeProps(buttonProps, focusProps, hoverProps)}
```

Comment:
a bit odd that the order here actually mattered, will dig to see why. Ideally it shouldn't rely on the props merge order

Reply:
funny enough, it's due to this line here https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/useFocus.ts#L78, it should be `undefined`
the reason the order mattered with it though is due to how mergeProps works with event handler merging, if you pass null as an event handler, it will remove ALL event handlers of that name

https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/utils/src/mergeProps.ts#L45 typeof null is 'object' so we won't go down this path
instead, we'll fall through here https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/utils/src/mergeProps.ts#L66 and overwrite everything

I'm going to push the change to this branch